src/dom-to-image-more.js

```diff
@@ -807,34 +859,35 @@
         }
 
         function width(node) {
-            var width = px(node, 'width');
+            const width = px(node, 'width');
 
-            if (isNaN(width)) {
-                const leftBorder = px(node, 'border-left-width');
-                const rightBorder = px(node, 'border-right-width');
-                width = node.scrollWidth + leftBorder + rightBorder;
-            }
-            return width;
+            if (!isNaN(width)) return width;
+
+            const leftBorder = px(node, 'border-left-width');
+            const rightBorder = px(node, 'border-right-width');
+            return node.scrollWidth + leftBorder + rightBorder;
         }
 
         function height(node) {
-            var height = px(node, 'height');
+            const height = px(node, 'height');
 
-            if (isNaN(height)) {
-                const topBorder = px(node, 'border-top-width');
-                const bottomBorder = px(node, 'border-bottom-width');
-                height = node.scrollHeight + topBorder + bottomBorder;
-            }
-            return height;
+            if (!isNaN(height)) return height;
+
+            const topBorder = px(node, 'border-top-width');
+            const bottomBorder = px(node, 'border-bottom-width');
+            return node.scrollHeight + topBorder + bottomBorder;
         }
 
         function px(node, styleProperty) {
-            let value = getComputedStyle(node).getPropertyValue(styleProperty);
-            if (value.slice(-2) !== 'px') {
-                return NaN;
+            if (node.nodeType === ELEMENT_NODE) {
```

Comment:
Only ask DOM Elements what their size is ... we can't ask a text node, comment node, etc...